src/components/modals/ImportWorkspaceFileModal.svelte

```diff
@@ -0,0 +1,104 @@
+<svelte:options immutable={true} />
+
+<script lang="ts">
+  import { createEventDispatcher } from 'svelte';
+  import * as Sidebar from '../../components/ui/Sidebar/index.js';
+  import type { User } from '../../types/app';
+  import type { Workspace, WorkspaceNodeEvent } from '../../types/workspace';
+  import type { WorkspaceTreeNode } from '../../types/workspace-tree-view';
+  import { joinPath, separateFilenameFromPath } from '../../utilities/workspaces.js';
+  import WorkspaceTreeView from '../workspace/WorkspaceTreeView/WorkspaceTreeView.svelte';
+  import Modal from './Modal.svelte';
+  import ModalContent from './ModalContent.svelte';
+  import ModalFooter from './ModalFooter.svelte';
+  import ModalHeader from './ModalHeader.svelte';
+
+  export let currentWorkspace: Workspace;
+  export let currentWorkspaceContents: WorkspaceTreeNode | null;
+  export let height: number = 400;
+  export let width: number = 380;
+  export let startingPath: string = '';
+  export let workspace: Workspace | null | undefined = null;
+  export let user: User | null;
+
+  const dispatch = createEventDispatcher<{
+    close: void;
+    confirm: { files: FileList; targetDirectory: string };
+  }>();
+
+  let targetPath: string = joinPath([currentWorkspace.name, startingPath]);
+  let targetDirectory: string = startingPath;
+  let targetFilename: string = '';
+  let saveButtonDisabled: boolean = false;
+  let filesToUpload: FileList;
+
+  $: {
+    const { filename, path } = separateFilenameFromPath(targetPath);
+    targetDirectory = path;
+    targetFilename = filename;
+  }
+  $: saveButtonDisabled = filesToUpload?.length === 0;
+
+  function onFolderClicked(event: CustomEvent<WorkspaceNodeEvent>) {
+    targetDirectory = event.detail.treeNodePath;
+  }
+
+  function save() {
+    if (!saveButtonDisabled) {
+      dispatch('confirm', {
+        files: filesToUpload,
+        targetDirectory: joinPath([
+          targetDirectory.replace(new RegExp(`^${currentWorkspace.name}`), '.'),
+          targetFilename,
+        ]),
+      });
+    }
+  }
+
+  function onKeydown(event: KeyboardEvent) {
+    const { key } = event;
+    if (key === 'Enter') {
+      event.preventDefault();
+      save();
+    }
+  }
+</script>
+
+<svelte:window on:keydown={onKeydown} />
+
+<Modal {height} {width}>
+  <ModalHeader on:close>Upload File(s) To Workspace</ModalHeader>
```

Comment:
We should use either "Import" or "Upload" files but not both. The menu items in workspace say "Import file".